build/source/engine/updateVars.f90

```diff
@@ -0,0 +1,631 @@
+     ! --> update the total water from the total water matric potential
+     case(iname_matLayer)
+      mLayerVolFracWatTrial(iLayer) = volFracLiq(mLayerMatricHeadTrial(ixControlIndex),vGn_alpha,theta_res,theta_sat,vGn_n,vGn_m)
+     ! --> update the total water matric potential (assume already have mLayerVolFracWatTrial given block above)
+     case(iname_liqLayer, iname_watLayer)
+      mLayerMatricHeadTrial(ixControlIndex) = matricHead(mLayerVolFracWatTrial(iLayer),vGn_alpha,theta_res,theta_sat,vGn_n,vGn_m) 
+     case default; err=20; message=trim(message)//'expect iname_lmpLayer, iname_matLayer, iname_liqLayer, or iname_watLayer'; return
+    end select
+   endif  ! if in the soil domain
+
+  endif  ! if hydrology state variable or uncoupled solution
+
+  ! initialize temperature 
+  select case(ixDomainType)
+   case(iname_veg);              xTemp = scalarCanopyTempTrial
+   case(iname_snow, iname_soil); xTemp = mLayerTempTrial(iLayer)
+   case default; err=20; message=trim(message)//'expect case to be iname_veg, iname_snow, iname_soil'; return
+  end select
+  
+  ! get iterations (set to maximum iterations if adjusting the temperature)
+  niter = merge(maxiter, 1, do_adjustTemp)
+
+  ! iterate
+  iterations: do iter=1,niter
+
+   ! -----
+   ! - compute derivatives...
+   ! ------------------------
+  
+   ! compute the derivative in total water content w.r.t. total water matric potential (m-1)
+   ! NOTE 1: valid for frozen and unfrozen conditions
+   ! NOTE 2: for case "iname_lmpLayer", dVolTot_dPsi0 = dVolLiq_dPsi
+   if(ixDomainType==iname_soil)then
+    select case( ixStateType(ixFullVector) )
+     case(iname_lmpLayer); dVolTot_dPsi0(ixControlIndex) = dTheta_dPsi(mLayerMatricHeadLiqTrial(ixControlIndex),vGn_alpha,0._dp,1._dp,vGn_n,vGn_m)*avPore
+     case default;         dVolTot_dPsi0(ixControlIndex) = dTheta_dPsi(mLayerMatricHeadTrial(ixControlIndex),vGn_alpha,theta_res,theta_sat,vGn_n,vGn_m)
+    end select
+   endif
+
+   ! compute the critical soil temperature below which ice exists
+   select case(ixDomainType)
+    case(iname_veg, iname_snow); Tcrit = Tfreeze
+    case(iname_soil);            Tcrit = crit_soilT( mLayerMatricHeadTrial(ixControlIndex) )
+    case default; err=20; message=trim(message)//'expect case to be iname_veg, iname_snow, iname_soil'; return
+   end select
+  
+   ! (compute the derivative in liquid water content w.r.t. temperature)
+   select case(ixDomainType)
+    case(iname_veg);  dTheta_dTkCanopy         = merge(dFracLiq_dTk(xTemp,snowfrz_scale)*scalarCanopyWat/(iden_water*canopyDepth),  0._dp, xTemp<Tcrit)
+    case(iname_snow); mLayerdTheta_dTk(iLayer) = merge(dFracLiq_dTk(xTemp,snowfrz_scale)*mLayerVolFracWatTrial(iLayer),             0._dp, xTemp<Tcrit)
+    case(iname_soil); mLayerdTheta_dTk(iLayer) = merge(dTheta_dTk(xTemp,theta_res,theta_sat,vGn_alpha,vGn_n,vGn_m),                 0._dp, xTemp<Tcrit) ! assume no volume expansion
```

Comment:
As far as I can tell, `function dTheta_dTk` was never called in previous versions - although this function does already exist in `soils_util.f90`. In that function, if `vGn_alpha < 0` and temperatures are above freezing, then the function takes a fractional power of a negative number and tries to store the result as a real(dp). 

I am not sure why it was allowing you to do this with gfortran ...